www/components/shelf/shelf.js

```diff
@@ -51,21 +51,25 @@ class shelf extends LitElement {
   }
 
   expandRoute(path) {
-    // find list item containing current window.location.pathname
-    let routeShelfListIndex = this.shelfList.findIndex(list => {
-      return list.path === path;
-    });
-
-    if (routeShelfListIndex > -1) {
-      this.shelfList[routeShelfListIndex].selected = true;
-      // force re-render
-      this.requestUpdate();
+    if (this.shelfList) {
+      // find list item containing current window.location.pathname
+      let routeShelfListIndex = this.shelfList.findIndex(list => {
+        return list.path === path;
+      });
+
+      if (routeShelfListIndex > -1) {
+        this.shelfList[routeShelfListIndex].selected = true;
+        // force re-render
+        this.requestUpdate();
+      }
     }
   }
 
   collapseAll() {
-    for (let i = 0; i < this.shelfList.length; i = i + 1) {
-      this.shelfList[i].selected = false;
+    if (this.shelfList) {
+      for (let i = 0; i < this.shelfList.length; i = i + 1) {
```

Comment:
I would recommend a `forEach` here.  

Also, could we just initialize `shelfList` to an empty array?  Then we could eliminate all the multiple checks
```javascript
if (this.shelfList) {
  // do stuff
}
```

Reply:
lol yes.

Reply:
I only use `map` if i need to return a new array.  Otherwise, `forEach` is my goto for non mutating changes.